meson.build

```diff
@@ -20,9 +20,9 @@ subdir('include/mt32emu') # public interface
 mt32emu_lib_includes = include_directories('include/mt32emu')
 mt32emu_dep_includes = include_directories('include')
 
-mt32emu_lib = library('mt32emu', mt32emu_sources,
-                      include_directories : mt32emu_lib_includes,
-                      cpp_args : ['-DMT32EMU_WITH_INTERNAL_RESAMPLER'])
+mt32emu_lib = static_library('mt32emu', mt32emu_sources,
```

Comment:
Unfortunately, this change has bad consequence :(

Packagers and end-users want static library, yes - but developers want dynamic library to keep the builds faster (otherwise Meson can't use dynamic linking optimization).

So:
- Pro: this change makes it a bit easier to use for packagers and end-users
- Con: this adds some tiny amount of time to linking time, and I do linking many, many times a day

Therefore remove this change, please.

Do I understand correctly - this change was needed because `meson setup -Ddefault_library=static`  in dosbox-staging repo didn't work for 0.53.0, but does work for >= 0.54.0?